core/src/com/unciv/logic/map/mapunit/UnitTurnManager.kt

```diff
@@ -19,7 +18,12 @@ class UnitTurnManager(val unit: MapUnit) {
         if (unit.currentMovement > 0
                 && unit.getTile().improvementInProgress != null
                 && unit.canBuildImprovement(unit.getTile().getTileImprovementInProgress()!!)
-        ) workOnImprovement()
+        ) {
+            val tile = unit.getTile()
+            tile.endTurn(unit)
+            tile.getCity()?.updateCitizens = true
```

Comment:
You are updating the citizens every time a worker works on an improvement in the city. Is this necessary?
Previously we only updated the citizens once the improvement was completed.